### PR TITLE
Accessibility in accordion panel

### PR DESCRIPTION
**Pull request by brichet**
This PR fixes an accessibility issue about accordion panel:
- removes the role `tab` to the title, which is not in a `tablist` element.
- adds an ID to the widgets if necessary, to correctly link the `aria-controls` attribute of the corresponding title.

### CODE REVIEW

**Comment by brichet** [comment]
> I guess you use the `Renderer` to reuse the `uuid`. But could you please prefer not changing any public API and add the `id` if it does not exist in `AccordionLayout.insertWidget` - inherited from https://github.com/jupyterlab/lumino/blob/main/packages/widgets/src/panellayout.ts#L87

Thanks for review @fcollonval

File: packages/widgets/tests/src/accordionpanel.spec.ts [hunk 1]
```diff
@@ -77,6 +77,37 @@ describe('@lumino/widgets', () => {
...
     });
 
+    describe('#accessibility()', () => {
+      it('should create a widget ID if it does not exist', () => {
+        let panel = new LogAccordionPanel();
+        let w1 = new Widget();
+        let w2 = new Widget();
+
+        // Expects a widget ID to be created.
+        expect(!w1.id);
```
Comment (fcollonval) [inline packages/widgets/tests/src/accordionpanel.spec.ts:87]: I think you cannot just call `expect`, you need to add an assertion like

`expect(w1.id).to.be.undefined;`

For test equality, you should use `expect(a).to.equal('value');`

You can look at the [chai library](https://www.chaijs.com/api/bdd/) to see what is available.

> Could you fix the test at line 123 that probably misguided you as it was missing an assertion.

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @brichet for the changes.

Could you add assertions to the tests and run `yarn run api` to update the API files (the associated CI job is failing).

File: packages/widgets/tests/src/accordionpanel.spec.ts (hunk 1 above)
Comment (brichet) [reply packages/widgets/tests/src/accordionpanel.spec.ts]: I wanted to make sure the tests pass :smile: 
Thanks for noticing.

**Review by fcollonval — APPROVED** [review]
Thanks @brichet

**Pull request merged**